tests/common/platform/transceiver_utils.py

```diff
@@ -133,13 +134,256 @@ def check_transceiver_dom_sensor_details(dut, asic_index, interfaces, xcvr_skip_
                     field, port_xcvr_dom_sensor["stdout"], intf)
 
 
-def check_transceiver_status(dut, asic_index, interfaces, xcvr_skip_list):
+def check_transceiver_status(dut, asic_index, interfaces, xcvr_skip_list, port_list_with_flat_memory):
     """
     @summary: Check transceiver information of all the specified interfaces in redis DB.
     @param dut: The AnsibleHost object of DUT. For interacting with DUT.
     @param interfaces: List of interfaces that need to be checked.
     """
     check_transceiver_basic(dut, asic_index, interfaces, xcvr_skip_list)
     check_transceiver_details(dut, asic_index, interfaces, xcvr_skip_list)
-    check_transceiver_dom_sensor_basic(dut, asic_index, interfaces, xcvr_skip_list)
-    check_transceiver_dom_sensor_details(dut, asic_index, interfaces, xcvr_skip_list)
+    check_transceiver_dom_sensor_basic(dut, asic_index, interfaces, xcvr_skip_list, port_list_with_flat_memory)
+    check_transceiver_dom_sensor_details(dut, asic_index, interfaces, xcvr_skip_list, port_list_with_flat_memory)
+
+
+def get_sfp_eeprom_map_per_port(eeprom_infos):
```

Comment:
@JibinBao You can modify this API to expect interfaces as an input instead of eeprom_infos. This will remove the overhead to the caller to send the EEPROM command output.
Internally, the `get_sfp_eeprom_map_per_port` API will run a for loop over the interfaces list and populate the dictionary with the command output for every port.

```
def get_sfp_eeprom_map_per_port(interfaces):
    for current_intf in interfaces:
        #Store eeprom information for current_intf to sfp_eeprom_map_per_port (via sfputil show eeprom -d -p current_intf). You can create a separate API as well at port level if needed.
```

Reply:
@mihirpat1, if so, we need to pass the argument of interfaces,  here we need to get all interfaces' information, so I think we don't need to pass the argements, to construct the argument of interfaces will take much more overhead.
In the other hand, getting the port information one by one (via sfputil show eeprom -d -p current_intf) will take much more time than getting them one time (via sfputil show eeprom -d), I thinke getting all port information one time will be better.

Reply:
I have do a test.
When port number is 28, it will take 25s to get the data one by one (via sfputil show eeprom -d -p current_intf)  .
Getting all ports' information will take about 4s one time (sfputil show eeprom -d ).